src/components/include/utils/threads/thread.h

```diff
@@ -104,23 +104,46 @@ class Thread : public QObject {
 #else
 class Thread {
 #endif
+
+  typedef enum ThreadCommandEnum {
```

Comment:
@orhan-mehmedov what for is this `typedef`? Why not just `enum`? We're not writing on C